Paper: Training a Supervised Cilia Segmentation Model from Self-SupervisionPaper added

salirezav opened the pull request:
If you are creating this PR in order to submit a draft of your paper, please name your PR with `Paper: <title>`. An editor will then add a `paper` label and GitHub Actions will be run to check and build your paper.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#preview-your-paper) for more information.

_Editor_: <!--editor--> Sanhita Joshi @sanhitamj  <!--end-editor-->

_Reviewers_: <!--reviewers-list--> 
- Chong Shen Ng @chongshenng 
- Tek Kshetri @iamtekson 
<!--end-reviewers-list-->

salirezav committed:
Paper added

salirezav committed:
DOIs added

rowanc1 commented on papers/Alireza_Vaezi/main.md:
```diff
@@ -0,0 +1,409 @@
+---
+# Ensure that this title is the same as the one in `myst.yml`
+Title: Training a Supervised Cilia Segmentation Model from Self-Supervision
```
```suggestion
title: Training a Supervised Cilia Segmentation Model from Self-Supervision
```

rowanc1 commented on papers/Alireza_Vaezi/main.md:
```diff
@@ -0,0 +1,409 @@
+---
+# Ensure that this title is the same as the one in `myst.yml`
+Title: Training a Supervised Cilia Segmentation Model from Self-Supervision
+Abstract: |
```
```suggestion
abstract: |
```

rowanc1 reviewed (commented):
Could you please lower case title and abstract! There are also a few references with missing DOIs!

salirezav committed:
title and abstract lowercase

ameyxd commented:
@salirezav Thanks for your submission. Can you update the DOIs that fail the checks? 

You may be able to add citation keys you want to ignore if their DOIs don't exist in myst.yml under error_rules:

  ```
error_rules:
    - rule: doi-exists
      severity: ignore
      keys:
        - abc
        - def01
```

salirezav committed:
doi keys added to error_rules

salirezav committed:
update

salirezav committed:
dois changed

salirezav commented:
Hi,

Thank you for your help. I made sure that every bibliography entry has a valid DOI, and otherwise appended them to the error_rule section. However, the check step still fails to complete.

salirezav committed:
minor change to force check to run

salirezav committed:
dois changed again

salirezav committed:
dois update

salirezav committed:
dois will work this time

salirezav committed:
webp images added

salirezav committed:
100th time is the charm

salirezav committed:
final

salirezav committed:
final

salirezav committed:
d2

iamtekson commented:
Hi @salirezav, I am Tek Kshetri, A graduate student at the University of Calgary, Canada. and one of the reviewers of your paper. I am currently reading your paper and will provide you with some comments here.

iamtekson commented on papers/Alireza_Vaezi/AR matrices.webp:
Looks like, this file is the same as `AR_matrices.png`. I suggest removing one of these.

salirezav replied:
All webp files were removed

iamtekson commented on papers/Alireza_Vaezi/main.md:
In this sentence, `Categorized into motile and primary, motile cilia can help the cell to propel, move the flow of fluid, or fulfill sensory functions, while primary cilia act as signal receivers, translating extracellular signals into cellular responses`, I suggest to add the reference.

salirezav replied:
Citation added.

iamtekson commented on papers/Alireza_Vaezi/main.md:
I suggest adding the citation to this paragraph too. As I am not from the medical field, I find things interesting. How do you know healthy cilia have regular and predictable movements and unhealthy cilia display a wide range of motion?

salirezav replied:
Citations added.

iamtekson commented on papers/Alireza_Vaezi/main.md:
Add full stop at the end of the paragraph `...constant throughout the video.`

salirezav replied:
Done.

iamtekson commented on papers/Alireza_Vaezi/main.md:
It would be better to explain why you selected $d = 5$ instead of the other number. Is it referenced from other literatures? or randomly selected?

salirezav replied:
Explanation added.

iamtekson commented on papers/Alireza_Vaezi/main.md:
Suggest adding more information in the figure caption, such as what these colors represent in the image. and what are the x and y-axis numbers?

salirezav replied:
I added a colormap in the plot and provided an explanation of the values.

iamtekson commented on papers/Alireza_Vaezi/main.md:
In this paragraph, `.......followed by a normalization of pixel intensities`, I suggest to add the normalization method such as, min-max normalization, standardization etc.

salirezav replied:
Done.

iamtekson commented on papers/Alireza_Vaezi/main.md:
Why are your validation and testing accuracy the same? My guess is you used a test dataset for the validation. If so, please remove validation accuracy from the table else please explain things.

chongshenng replied:
I agree with @iamtekson's comments - the test scores stood out to me since it is quite peculiar to have the same test and validation scores.

salirezav replied:
I fixed this issue.

iamtekson commented on papers/Alireza_Vaezi/main.md:
I found this section little bit incomplete. I suggest adding a table with hyperparameters and their values for a few of the hyperparameters such as, optimizer, loss function name, number of training/test/validation samples, callback function names (if any) etc.

chongshenng replied:
I agree with @iamtekson. These information will also be useful for reproducibility and help increase visibility of your work.

salirezav replied:
I appended a table including the details of the model and data.

iamtekson commented on papers/Alireza_Vaezi/main.md:
suggest to remove this section as it is not the part of the paper.

salirezav replied:
Done!

iamtekson reviewed (commented):
Overall, the paper is well-written and easy to understand, even for someone not from the medical field. I appreciate the clarity and thoroughness of the authors' explanations. Thank you for giving me the opportunity to review this work.

# Summary of paper:
The study introduces a self-supervised framework for automating cilia segmentation, crucial for diagnosing ciliopathies. Traditional methods rely on labor-intensive, manually labeled data. The new approach leverages optical flow to generate motion-based pseudolabels from healthy cilia videos, creating a robust training set for a semi-supervised neural network. This model effectively segments both motile and immotile cilia, overcoming inconsistencies in ciliary motion. Using a two-stage process involving optical flow and autoregressive modeling, the method achieves high accuracy without expert-drawn masks, significantly enhancing automated cilia analysis and potentially accelerating research and diagnostics for ciliopathies.

# My general comments
I have listed few general comments as below,

1. All the figures have two copies in `png` and `webp` formats. I suggest removing one of them.
2. The paper could be further enhanced by including a detailed discussion section. This section could compare the proposed method with existing approaches, elaborate on potential limitations, and suggest future research directions. Furthermore, it would be nice to explain about the generalization of the model. Additionally, a deeper analysis of the pseudolabeling accuracy and its impact on the model’s performance would provide more comprehensive insights into the framework’s efficacy.

chongshenng commented:
Hello @salirezav 👋 , thanks for your submission! I'm Chong Shen, one of the reviewers of your paper. I'll add my comments shortly.

chongshenng commented on papers/Alireza_Vaezi/main.md:
I like how clear the sentences are. To improve readability and flow, I suggest using words and phrases to link to the previous sentences. E.g.: 

> ... sweep rhythmically to transport substances. _When cilia become dysfunctional,_ they are indicative of diseases that can disrupt organs such as the lungs and kidneys. _Therefore,_ it is essential to understand cilia behaviour in diagnosing and treating such diseases. 

and 

> _But,_ the tasks of automatically analysing cilia are often labor and time-intensive, since there is a lack of automated segmentation. _In this work, we overcome this significant bottleneck_ by developing a robust, self-supervised framework ...

salirezav replied:
Resolved.

chongshenng commented on papers/Alireza_Vaezi/main.md:
Since this paragraph is about unsupervised learning, it would be instructive to readers to clarify that your approach of computing the motion vectors is unsupervised. Consider rephrasing to:

> ... accurate segmentation masks for cilia. _Inspired by advances in unsupervised methods for image segmentation, in this work, we firstly compute the motion vectors using optical flow of the ciliary regions and then apply autoregressive modelling to capture their temporal dynamics. Autoregressive modelling is advantageous since the labels are features themselves._ By analyzing the optical flow vectors, ...

salirezav replied:
Thanks for the help. Resolved.

chongshenng commented on papers/Alireza_Vaezi/main.md:
Since this is the first time readers would see an image of a cilia, it would be useful to annotate a region occupied by the cilia or describe in words.

salirezav replied:
Resolved!

chongshenng commented on papers/Alireza_Vaezi/main.md:
I couldn't find a reference in the text to figures (1) and (2). Perhaps it'd be more useful to create combine all figures (1) to (3) into one, and then add a subfigure (a) to (c). Then, your reference could be updated to something along the lines of:

> `@fig:sample_OF` show a sample of the flow region around a cilia. The raw image is shown in (a) and a manually labeled ground truth in (b). The vorticity computed from the optical flow from (b) is shown in (c).

salirezav replied:
I replaced them with one figure.

chongshenng commented on papers/Alireza_Vaezi/main.md:
It's unclear to me here what you're modelling. You mention 'rotation', but later refer to the temporal evolution of vectors, which is not intuitive. As I understand, rotation (in this case, a 2D rotation) would be vorticity and the vector is orthogonal to the page. Are you modelling the vorticity, velocity gradients, or the velocities themselves? Can you please revise this paragraph?

salirezav replied:
I added more detail to the OF and AR sections.

chongshenng commented on papers/Alireza_Vaezi/main.md:
Here, you mentioned "improved performance in accurately detecting ciliary regions". Can you clarify in the text what the improvement is compared relative to? I don't recall reading about a baseline comparison in the text.

salirezav replied:
I modified this paragraph.

chongshenng commented on papers/Alireza_Vaezi/main.md:
It's not clear to me what are the training images. Is it (d)?

chongshenng commented:
## General comments
The authors propose a two-stage process to improve image segmentation models for cilia detection. The first step of the process first applies a series of image processing techniques, followed by auto regressive modelling. Finally, they apply a FPN architecture to train their image segmentation model. 

The results from this work is promising especially since the proposed method appears to be quite novel. I've provided a few comments in the manuscript. However, there're still a few points below that I'd encourage the authors to address before accepting it:

* There's no description of the cilia video/image dataset in this paper. Can you provide a brief description of it in the `Methodology` section? Is it an open source dataset? What were the training data sizes?
* It'd be instructive to readers if a simple diagram can be included to show the workflow from raw video/image to a trained model. 
* The results refer to IoU, sensitivity, and specifity, which are fine. It would be more instructive to readers if there's an indication of a baseline for comparison of performance.

sanhitamj commented on papers/Alireza_Vaezi/main.md:
Need the full form of OF when used first.

salirezav replied:
Resolved.

chongshenng reviewed (changes requested):
Please refer to my commment [here](https://github.com/scipy-conference/scipy_proceedings/pull/929#issuecomment-2187313706).

salirezav commented:
> Hi @salirezav, I am Tek Kshetri, A graduate student at the University of Calgary, Canada. and one of the reviewers of your paper. I am currently reading your paper and will provide you with some comments here.

Hello @iamtekson,
Thank you for your comments! I will be addressing your suggestions and comments in the following days.

salirezav commented:
> Hello @salirezav 👋 , thanks for your submission! I'm Chong Shen, one of the reviewers of your paper. I'll add my comments shortly.

Hello @chongshenng,
Thank you for your comments and suggestions! I will be addressing the suggestions throughout the following days.

salirezav committed:
removed webp images

salirezav committed:
comments addressed 01

salirezav committed:
updates July 7

salirezav committed:
updates 2

salirezav committed:
fin

salirezav committed:
slight change

salirezav committed:
conclusion

salirezav committed:
final updates

sanhitamj commented:
Hi @salirezav Just a reminder that 2nd Sept is the final deadline for the author(s) to make edits to the submission.

salirezav commented:
> Hi @salirezav Just a reminder that 2nd Sept is the final deadline for the author(s) to make edits to the submission.

Hi @sanhitamj ! Thank you for the reminder! All of the changes suggested by the reviewers have been already made and pushed. I only need to make a minor change and then everything will be ready way before the deadline.

Thank you for all the help!

sanhitamj commented:
@salirezav thanks for the submission and the subsequent edits as well.

@iamtekson and @chongshenng thank you so much for the careful review. Will you please get back with your final decision about the paper by 9th Sept?

salirezav committed:
paper last update

iamtekson commented:
Now, it looks much more improved. I vote for accepting it. Thanks @sanhitamj and @salirezav for providing me the opportunity to review it.
